cloudformation/slack-router-cfn-template.js

```diff
@@ -0,0 +1,94 @@
+const cf = require('@mapbox/cloudfriend')
+
+const Resources = {
+  SlackRouterApi: {
+    Type: 'AWS::ApiGatewayV2::Api',
+    Properties: {
+      Name: 'slack-router-api',
+      ProtocolType: 'HTTP',
+    },
+  },
+  SlackToLambdaIntegration: {
+    Type: 'AWS::ApiGatewayV2::Integration',
+    Properties: {
+      ApiId: cf.ref('SlackRouterApi'),
+      Description: 'Integration for Slack Slash Commands',
+      IntegrationType: 'AWS_PROXY',
+      IntegrationUri: cf.join('', [
+        'arn:',
+        cf.partition,
+        ':apigateway:',
+        cf.region,
+        ':lambda:path/2015-03-31/functions/',
+        cf.getAtt('SlackRouterLambda', 'Arn'),
+        '/invocations',
+      ]),
+      PayloadFormatVersion: '2.0',
+    },
+  },
+  SlackToLambdaRoute: {
+    Type: 'AWS::ApiGatewayV2::Route',
+    Properties: {
+      ApiId: cf.ref('SlackRouterApi'),
+      AuthorizationType: 'NONE',
+      RouteKey: 'POST /slack-to-lambda',
+      Target: cf.join('/', [
+        'integrations',
+        cf.ref('SlackToLambdaIntegration'),
+      ]),
+    },
+  },
+  SlackToLambdaDeploy: {
+    Type: 'AWS::ApiGatewayV2::Deployment',
+    DependsOn: 'SlackToLambdaRoute',
+    Properties: {
+      ApiId: cf.ref('SlackRouterApi'),
+    },
+  },
+  SlackRouterApiProdStage: {
+    Type: 'AWS::ApiGatewayV2::Stage',
+    Properties: {
+      ApiId: cf.ref('SlackRouterApi'),
+      AutoDeploy: 'true',
+      DeploymentId: cf.ref('SlackToLambdaDeploy'),
+      StageName: 'prod',
+    },
+  },
+  SlackRouterLambda: {
+    Type: 'AWS::Lambda::Function',
+    Properties: {
+      FunctionName: 'slack-router-lambda',
+      Handler: 'slack-router-lambda.handler',
+      Role: cf.sub(
+        'arn:aws:iam::${AWS::AccountId}:role/service-role/test-api-lambda-sns-role-6ndj69lt'
+      ),
+      Code: {
+        S3Bucket: 'lambda-andria',
+        S3Key: 'slack-router-lambda.zip',
+      },
```

Comment:
Accept the S3 bucket and key as a parameter, and reference it here.

Reply:
I was initially planning on putting the gitsha as a parameter which will then be added to the S3Key to complete it—since I thought it was only the name of the zip file (which will be the gitsha) that will be changing once the automated deployment is setup—like so:
```
const Parameters = {
  LambdaGitSha: {
    Type: 'String',
    Description: 'Git SHA of latest commit for Lambda function',
  },
}
...
Code: {
          S3Bucket: 'stork-us-east-1',
          S3Key: cf.sub('bundles/slack-bots/${LambdaGitSha}.zip'),
      },
```
Do we instead want both the S3Bucket and the full S3Key values to be parameters?

Reply:
Great question. We need to have the bucket and prefix information configurable for both security/flexibility.

Reply:
a43f0d8d95895

Added bucket name and gitsha as parameters.